pages/index.tsx

```diff
@@ -1,19 +1,180 @@
-import dynamic from "next/dynamic";
-
-const Category = dynamic(() => import("../components/category/Category"));
-
-const Home = () => {
-  return (
-    <div>
-      {/* <Carousel /> */}
-      {/* <Benefits /> */}
-      {/* <Offers /> */}
-      <Category />
-      {/* <Newest /> */}
-      {/* <Banners /> */}
-      {/* <Brands /> */}
-    </div>
-  );
+import React, { Profiler, useEffect, useState } from 'react'
+import dynamic from 'next/dynamic';
+import BottomModal from '../components/BottomModal';
+import OptionCard from '../components/Map/OptionCard';
+import { GiHospital } from 'react-icons/gi';
+import { optionData } from '../components/Map/StoreMarkerData';
+import { useRouter } from 'next/router'
+ 
+
+
+
+type Coords = {
+	lat: number;
+	long: number;
 };
 
-export default Home;
+import MapSearch from '../components/Map/MapSearch';
+
+const homepage = () => {
+
+	const MapWithNoSSR = dynamic(() => import("../components/Map"), {
+		ssr: false
+	});
+
+	// create a state value called query in typescript
+	const [query, setQuery] = useState<string>("");
+	const [coords, setCoords] = useState<Coords>({ lat: 46.603354, long: 1.8883335 });
+	const [isOptionModalOpen, setIsOptionModalOpen] = useState<boolean>(false);
+	const [isOptionDetailOpen, setIsOptionDetailOpen] = useState<boolean>(false);
+	const [option, setOption] = useState<{ tagName: string; tagValue: string; }>({
+		tagName: "amenity",
+		tagValue: "restaurant"
+	});
+	const [selectedStore, setSelectedStore] = useState<any>(null);
+	const router = useRouter();
+
+
+
+	//TODO Pseudo loading for now. Need to figure out to do this using map load events
+	const [mapLoading,setMapLoading] = useState<boolean>(false);
+
+
+	useEffect(()=>{
+		setMapLoading(true);
+		setTimeout(()=>{
+			setMapLoading(false)
+		},1000)
+
+	},[query])
+
+
+
+	 const handleModalOpen = () => {
+		setIsOptionModalOpen((prevState) => (!prevState));
+			 }
+
+	const handleModalClose = () => {
+		setIsOptionModalOpen(false);
+	}
+
+
+	const handleOptionDetailOpen = () => {
+		setIsOptionDetailOpen((prevState) => (!prevState));
+	}
+
+	const handleOptionDetailClose = () => {
+		setIsOptionDetailOpen(false);
+	}
+
+
+	const fetchLocationByQuery = (query: string) => {
+		let url = "https://nominatim.openstreetmap.org/search?format=jsonv2" +
+			"&q=" + query;
+		fetch(url, {
+			method: "GET",
+			mode: 'cors',
+			headers: {
+				"Access-Control-Allow-Origin": "https://o2cj2q.csb.app"
+			}
+		})
+			.then((response) => response.json())
+			.then((data) => {
+				setCoords({ lat: data[0].lat, long: data[0].lon })
+			});
+
+	}
+
+	const fetchStoresByLocation = (lat: number, long: number, tagValue: string, tagName: string) => {
+
+		// using fetch because axios showing cors error. Discuss with team
+		let url = `https://api.osm-dev.becknprotocol.io/stores?tagName=${tagName}&tagValue=${tagValue}&latitude=${lat}&longitude=${long}`;
```

Comment:
Hi @ankitShogun - can we pull the base url from .env file as I believe the base url will change based on environment?
@nikhil2128 your thoughts?

Reply:
@richapgupta yeah added in the env but forgot to use it. Updated the pr.